src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -96,6 +99,94 @@ module.exports = function(scheduleUpdate : (fiber: Fiber, priorityLevel : Priori
+  function checkClassInstance(workInProgress: Fiber, inst: any) {
+    if (__DEV__) {
+      const name = getName(workInProgress, inst);
+      const renderPresent = inst.render;
+      warning(
+        renderPresent,
+        '%s(...): No `render` method found on the returned component ' +
+        'instance: you may have forgotten to define `render`.',
+        name
+      );
+      const noGetInitialStateOnES6 = (
+        !inst.getInitialState ||
+        inst.getInitialState.isReactClassApproved
+      );
+      warning(
+        noGetInitialStateOnES6,
+        'getInitialState was defined on %s, a plain JavaScript class. ' +
+        'This is only supported for classes created using React.createClass. ' +
+        'Did you mean to define a state property instead?',
+        name
+      );
+      const noGetDefaultPropsOnES6 = (
+        !inst.getDefaultProps ||
+        inst.getDefaultProps.isReactClassApproved
+      );
+      warning(
+        noGetDefaultPropsOnES6,
+        'getDefaultProps was defined on %s, a plain JavaScript class. ' +
+        'This is only supported for classes created using React.createClass. ' +
+        'Use a static property to define defaultProps instead.',
+        name
+      );
+      const noInstancePropTypes = !inst.propTypes;
+      warning(
+        noInstancePropTypes,
+        'propTypes was defined as an instance property on %s. Use a static ' +
+        'property to define propTypes instead.',
+        name,
+      );
+      const noInstanceContextTypes = !inst.contextTypes;
+      warning(
+        noInstanceContextTypes,
+        'contextTypes was defined as an instance property on %s. Use a static ' +
+        'property to define contextTypes instead.',
+        name,
+      );
+      const noComponentShouldUpdate = typeof inst.componentShouldUpdate !== 'function';
+      warning(
+        noComponentShouldUpdate,
+        '%s has a method called ' +
+        'componentShouldUpdate(). Did you mean shouldComponentUpdate()? ' +
+        'The name is phrased as a question because the function is ' +
+        'expected to return a value.',
+        name
+      );
+      const noComponentDidUnmount = typeof inst.componentDidUnmount !== 'function';
+      warning(
+        noComponentDidUnmount,
+        '%s has a method called ' +
+        'componentDidUnmount(). But there is no such lifecycle method. ' +
+        'Did you mean componentWillUnmount()?',
+        name
+      );
+      const noComponentWillRecieveProps = typeof inst.componentWillRecieveProps !== 'function';
+      warning(
+        noComponentWillRecieveProps,
+        '%s has a method called ' +
+        'componentWillRecieveProps(). Did you mean componentWillReceiveProps()?',
+        name
+      );
+    }
+
+    if (inst.state && (typeof inst.state !== 'object' || isArray(inst.state))) {
+      // TODO: Change this to be a warning.
+      throw new Error(`${getName(workInProgress, inst)}.state: must be set to an object or null`);
```

Comment:
Can you please use invariant module for this? Otherwise the error text won't be automatically stripped out in production builds.